tangent/tf_extensions.py

```diff
@@ -238,7 +238,7 @@ def dtfreduce_mean(y, x, axis=None, keep_dims=False):
 
 @adjoint(tf.reduce_max)
 def dtfreduce_max(y, x, axis=None, keep_dims=False):
-  mask = tf.to_float(
+  mask = tf.cast(
```

Comment:
You'll need to replace `tf.to_float(.)` with `tf.cast(., tf.float32)`. Check the tests, they should tell you when things are complete.